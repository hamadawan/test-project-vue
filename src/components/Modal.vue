/StudentsRegistration.vue
<template>
  <div
    v-if="show"
    class="modal fixed w-full h-full top-0 left-0 flex items-center justify-center"
  >
    <div
      @click.self="$emit('close')"
      class="modal-overlay absolute w-full h-full bg-gray-900 opacity-50"
    ></div>

    <div
      class="modal-container bg-white w-11/12 md:max-w-md mx-auto rounded shadow-lg z-50 overflow-y-auto"
    >
      <div class="modal-content py-4 text-left px-6">
        <div class="flex justify-center items-center pb-3">
          <p class="text-2xl font-bold text-center">{{ content }}</p>
        </div>
        <div class="flex items-center justify-end border-t border-teal-500 py-2">
          <button
            class="flex-shrink-0 mr-2 border-black border-2 text-teal-500 hover:text-black-800 text-sm py-1 px-2 rounded"
            type="button"
            @click="$emit('close')"
          >
            Cancel
          </button>
          <button
            class="flex-shrink-0 bg-black hover:bg-teal-700 border-black hover:border-black-700 text-sm border-2 text-white py-1 px-2 rounded"
            type="button" 
            @click="$emit('confirm')"
          >
            Okay
          </button>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: "StudentsRegistration",
  props: ['show', 'content']
};
</script>
