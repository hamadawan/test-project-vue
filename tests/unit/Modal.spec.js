import { shallowMount, mount } from '@vue/test-utils'
import Modal from '@/components/Modal.vue'

describe('ProductForm.vue', () => {
  it('Renders props.content when passed', () => {
    const msg = 'Product created successfully!'
    const wrapper = shallowMount(Modal, {
      propsData: { content: msg, show: true }
    })
    expect(wrapper.text()).toMatch(msg)
  })

  it('Renders correctly', () => {
    const wrapper = mount(Modal)
    expect(wrapper.element).toMatchSnapshot()
  })
})