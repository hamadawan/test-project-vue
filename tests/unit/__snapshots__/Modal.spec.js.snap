// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductForm.vue Renders correctly 1`] = `<!---->`;
